src/main/java/com/rtb/projectmanagementtool/taskblocker/TaskBlockerController.java

```diff
@@ -2,17 +2,99 @@
 
 import com.google.appengine.api.datastore.*;
 import com.google.appengine.api.datastore.Query.*;
+import com.rtb.projectmanagementtool.task.*;
+import com.rtb.projectmanagementtool.task.TaskData.Status;
+import java.util.ArrayList;
 import java.util.HashSet;
 
 /** Class controlling the TaskBlockerData object. */
 public final class TaskBlockerController {
-  public TaskBlockerController() {}
+  private DatastoreService datastore;
+  private TaskController taskController;
+  private Filter NO_FILTER = null;
 
-  public HashSet<TaskBlockerData> getTaskBlockers(DatastoreService datastore, long taskID) {
+  public TaskBlockerController(DatastoreService datastore, TaskController taskController) {
+    this.datastore = datastore;
+    this.taskController = taskController;
+  }
+
+  // Add methods
+
+  public void addTaskBlocker(long taskID, long blockerID) {
+    addTaskBlocker(new TaskBlockerData(taskID, blockerID));
+  }
+
+  public void addTaskBlocker(TaskBlockerData taskBlocker) {
+    // Get task
+    TaskData task;
+    try {
+      task = taskController.getTaskByID(taskBlocker.getTaskID());
+    } catch (NullPointerException e) {
+      System.out.println("Cannot find task with provided taskID");
+      return;
+    }
+    // Get blocker
+    TaskData blocker;
+    try {
+      blocker = taskController.getTaskByID(taskBlocker.getBlockerID());
+    } catch (NullPointerException e) {
+      System.out.println("Cannot find task with provided blockerID");
+      return;
+    }
+    // Add taskBlocker
+    if (task.getStatus() != Status.COMPLETE) {
+      TransactionOptions options = TransactionOptions.Builder.withXG(true);
+      Transaction transaction = datastore.beginTransaction(options);
+      try {
+        datastore.put(taskBlocker.toEntity());
+        assert containsCycle() == false;
+        transaction.commit();
+      } finally {
+        if (transaction.isActive()) {
+          transaction.rollback();
+        }
+      }
+    } else {
+      System.out.println("Cannot block a task set to COMPLETE.");
+    }
+  }
+
+  private boolean containsCycle() {
+    // TODO: Implement cycle detection algorithm
+    return false;
+  }
+
+  // Get methods
+
+  public HashSet<TaskBlockerData> getAllTaskBlockers() {
+    return getTaskBlockers(NO_FILTER);
+  }
+
+  public ArrayList<TaskData> getTaskBlockerTasks(long taskID) {
+    HashSet<TaskBlockerData> taskBlockers = getTaskBlockers(taskID);
+    ArrayList<TaskData> tasks = new ArrayList<>();
+    for (TaskBlockerData taskBlocker : taskBlockers) {
+      tasks.add(taskController.getTaskByID(taskBlocker.getBlockerID()));
```

Comment:
Queries in loops could get expensive.  Maybe look into building a list of keys and doing 1 big query.